File: Frontend/skillshare_frontend/lib/models/review.dart
class Review {
  final int id;
  final int rating;
  final String? comment;
  final int reviewerId;
  final String reviewerFullName;
  final int reviewedUserId;
  final int? requestId;
  final DateTime createdAt;

  const Review({
    required this.id,
    required this.rating,
    required this.comment,
    required this.reviewerId,
    required this.reviewerFullName,
    required this.reviewedUserId,
    required this.requestId,
    required this.createdAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: (json['id'] as num).toInt(),
      rating: (json['rating'] as num).toInt(),
      comment: json['comment'] as String?,
      reviewerId: (json['reviewerId'] as num).toInt(),
      reviewerFullName: (json['reviewerFullName'] ?? '') as String,
      reviewedUserId: (json['reviewedUserId'] as num).toInt(),
      requestId: (json['requestId'] as num?)?.toInt(),
      createdAt: DateTime.parse((json['createdAt'] ?? DateTime.now().toUtc().toIso8601String()) as String),
    );
  }
}

File: Backend/src/main/java/com/skillshare/model/Review.java
package com.skillshare.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "reviews")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "reviewer_id", nullable = false)
    private User reviewer;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "reviewed_user_id", nullable = false)
    private User reviewedUser;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "request_id")
    private SkillRequest request;

    @Column(nullable = false)
    private int rating;

    @Column(length = 2000)
    private String comment;

    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @PrePersist
    void onCreate() {
        createdAt = Instant.now();
    }
}
